src/hal/src/lib.rs

```diff
@@ -304,7 +305,9 @@ pub trait Instance {
     fn enumerate_adapters(&self) -> Vec<Adapter<Self::Backend>>;
 }
 
-/// Different types of a specific API.
+/// The `Backend` trait wraps together all the types needed
+/// to implement a graphics backend.  Each backend, such as OpenGL
```

Comment:
the second sentence seems very repetit-ory. Perhaps, something like this would sound better?
> The `Backend` trait wraps together all the types needed to implement a graphics backend, turning into concrete types on OpenGL, Metal, etc.

Reply:
Yeah, it could be phrased better.  This is one of the Important Bits though, so I feel like the repetition is useful to make sure the relationship between the trait, the backend modules, and the backend API's are clear.  I'll rework it.